cannon/mipsevm/iface.go

```diff
@@ -56,6 +56,9 @@ type FPVMState interface {
 	// EncodeWitness returns the witness for the current state and the state hash
 	EncodeWitness() (witness []byte, hash common.Hash)
 
+	// EncodeProof returns the proof for the current state
+	EncodeProof() []byte
```

Comment:
I probably wouldn't modify the main interface for this - in particular because the tests here need an "initial"/incomplete proof without the memory proofs that are generated by running FPVM.step().   

I'd suggest adding a helper function in the test utils [here](https://github.com/ethereum-optimism/optimism/blob/cd1cc1039b58ce5146b49d1a018bc303cdaa56b2/cannon/mipsevm/tests/helpers.go#L53-L59).

Reply:
solved and replaced `EncodeProof`  with `ThreadProofEncoder`